core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -621,19 +744,34 @@ object ConsumerGroupCommand extends Logging {
       }
     }
 
-    def exportOffsetsToReset(assignmentsToReset: Map[TopicPartition, OffsetAndMetadata]): String = {
-      val rows = assignmentsToReset.map { case (k,v) => s"${k.topic},${k.partition},${v.offset}" }(collection.breakOut): List[String]
-      rows.foldRight("")(_ + "\n" + _)
+    def exportOffsetsToReset(assignmentsToReset: Map[String, Map[TopicPartition, OffsetAndMetadata]]): String = {
+      val isSingleGroupQuery = opts.options.valuesOf(opts.groupOpt).size() == 1
+      val csvWriter =
+        if (isSingleGroupQuery) CsvUtils().writerFor[CsvRecordNoGroup]
+        else CsvUtils().writerFor[CsvRecordWithGroup]
+      val rows = assignmentsToReset.flatMap { case (groupId, partitionInfo) =>
+        partitionInfo.map { case (k: TopicPartition, v: OffsetAndMetadata) =>
+          val csvRecord =
+            if (isSingleGroupQuery) CsvRecordNoGroup(k.topic, k.partition, v.offset)
+            else CsvRecordWithGroup(groupId, k.topic, k.partition, v.offset)
+          csvWriter.writeValueAsString(csvRecord)
+        }(collection.breakOut): List[String]
+      }
+      val csv = rows.foldRight("")(_ + _)
+      csv
```

Comment:
> Unnecessary val definition?

Nope, intentionally. Just for the purpose of better readability and clarity. Do you think it's worth removing?

Reply:
IMHO defining a variable just for the sake of returning it is excessive. I haven't seen that being exercised in the scope of this project either.

Reply:
Perhaps turn the right-hand side into a `def` with a nice name?

Reply:
@hachikuji @vahidhashemian renamed method to `exportOffsetsToCsv` and dropped val. Is it ok? 

> * In the below example the last line seems to be unnecessary
> 
> ```
> stack@devstack1:~/kafka$ bin/kafka-consumer-groups.sh --bootstrap-server localhost:9092 --reset-offsets --all-groups
> 
> WARN: No action will be performed as the --execute option is missing.In a future major release, the default behavior of this command will be to prompt the user before executing the reset rather than doing a dry run. You should add the --dry-run option explicitly if you are scripting this command and want to keep the current default behavior without prompting.
> Error: Assignments can only be reset if the group 'g1' is inactive, but the current state is Stable.
> Error: Assignments can only be reset if the group 'g2' is inactive, but the current state is Stable.
> 
> GROUP                          TOPIC                          PARTITION  NEW-OFFSET
> ```
> Also in this example, if a group is inactive, it would report an error saying there are missing arguments. Not sure if this is inherited from the existing code, but it would probably provide a better user experience if the above command also complained about the missing argument (before checking the group status). Validation of the command should all happen before executing its logic.

Unable to reproduce. After latest changes the command generates the following output:
```
One of the reset scopes should be defined: --all-topics, --topic.
```

Reply:
I can still reproduce these issues:

1. Start a consumer in group `g1` and then run `bin/kafka-consumer-groups.sh --bootstrap-server localhost:9092 --reset-offsets --all-groups`. What I get is below (the last line --table headers-- is still there):
```
Error: Assignments can only be reset if the group 'g1' is inactive, but the current state is Stable.

GROUP                          TOPIC                          PARTITION  NEW-OFFSET
```

2. Now stop the consumer above, and run the same command. I get
```
One of the reset scopes should be defined: --all-topics, --topic.
```
Which sort of contradicts the previous try. If there is an issue with command arguments the previous try should have caught it too.

I hope I was clear, and did not miss something. Thanks!

Reply:
@vahidhashemian that's weird, still unable to reproduce. Here are my steps and results:

# Starting consumer group g1:
```
$ bin/kafka-consumer-groups.sh --bootstrap-server localhost:9092 --consumer.config config/consumer.properties --topic test
```

# Ensuring it's on the list and is functioning
```
$ bin/kafka-consumer-groups.sh --list --bootstrap-server localhost:9092
g1
```

```
$ bin/kafka-consumer-groups.sh --describe --group g1 --bootstrap-server localhost:9092
TOPIC           PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID                                     HOST            CLIENT-ID       GROUP
test            0          0               0               0               consumer-1-fc6b81cd-3c66-479b-91d0-941a595b4edf /172.29.137.160 consumer-1      g1
```

# Running command 
```
$ bin/kafka-consumer-groups.sh --bootstrap-server localhost:9092 --reset-offsets --all-groups
WARN: No action will be performed as the --execute option is missing.In a future major release, the default behavior of this command will be to prompt the user before executing the reset rather than doing a dry run. You should add the --dry-run option explicitly if you are scripting this command and want to keep the current default behavior without prompting.
Missing required argument "[group]"
Option                                  Description
------                                  -----------
--all-groups                            Apply to all consumer groups.
```


# Stopping consumer group
```
$ bin/kafka-consumer-groups.sh --list --bootstrap-server localhost:9092
g1
^C
Processed a total of 0 messages
```

#Running this command again:
```
$ bin/kafka-consumer-groups.sh --bootstrap-server localhost:9092 --reset-offsets --all-groups
WARN: No action will be performed as the --execute option is missing.In a future major release, the default behavior of this command will be to prompt the user before executing the reset rather than doing a dry run. You should add the --dry-run option explicitly if you are scripting this command and want to keep the current default behavior without prompting.
Missing required argument "[group]"
Option                                  Description
------                                  -----------
--all-groups                            Apply to all consumer groups.
```

Still unable to reproduce...

Reply:
Hmm. I get this at "Running command" step above:
```
$ bin/kafka-consumer-groups.sh --bootstrap-server localhost:9092 --reset-offsets --all-groups
WARN: No action will be performed as the --execute option is missing.In a future major release, the default behavior of this command will be to prompt the user before executing the reset rather than doing a dry run. You should add the --dry-run option explicitly if you are scripting this command and want to keep the current default behavior without prompting.

Error: Assignments can only be reset if the group 'g1' is inactive, but the current state is Stable.

GROUP                          TOPIC                          PARTITION  NEW-OFFSET     
```
The rest is the same. Not sure what's causing this inconsistency. I checked out the code from your branch for this PR.

Reply:
@vahidhashemian thanks, you were right 👍 rebuilt from scratch and fixed now. Had to pay attention at the cmd syntax